tests/test-sdi-progress-window.c

```diff
@@ -0,0 +1,318 @@
+     "and then "
+     "right-to-left softly)\nThe YES and NO buttons will be disabled for 5 "
+     "seconds, so wait at least until they are re-enabled.",
+     "Confirm that is correct.", test_pulse_bar},
+    {3, "/progress_window/test_close_bar",
+     "The window must disappear and no new window should appear (check below "
+     "this window just in case).\n",
+     "Confirm that is correct.", test_close_bar},
+    {4, "/progress_window/test_manual_hide",
+     "A window with a progress bar must appear with the name and icon of a "
+     "snapped application.",
+     "Click on the 'Hide' button. The window must "
+     "disappear and no new window should appear (check below this one). "
+     "Confirm that is correct.",
+     test_manual_hide},
+    {5, "/progress_window/test_dual_progress_bar_1",
+     "A window with a progress bar must appear with the name and icon of a "
+     "snapped application, growing from 1 to 10 and going back to 1 "
+     "periodically. It must not 'pulse' (move left-to-right and then "
+     "right-to-left softly).\n"
+     "The text in the progress bar should be 'Description for task X (X/10)', "
+     "being X a number between 1 and 10.\nIf no window is shown, check if it "
+     "is below this window.",
+     "Confirm that is correct.", test_dual_progress_bar1},
+    {6, "/progress_window/test_dual_progress_bar_2",
+     "A second progress bar must appear in the same window with the name and "
+     "icon of a "
+     "snapped application, growing from 1 to 10 and going back to 1 "
+     "periodically. It must not 'pulse' (move left-to-right and then "
+     "right-to-left softly).\n"
+     "The text in the progress bar should be 'Description for task X (X/10)', "
+     "being X a number between 1 and 10.\nIf no window is shown, check if it "
+     "is below this window.",
+     "Confirm that is correct.", test_dual_progress_bar2},
+    {7, "/progress_window/test_dual_progress_bar_3",
+     "The first progress bar must disappear, remaining only the second "
+     "progress bar. Neither a new window, nor a new progress bar, should "
+     "appear (check below "
+     "this window just in case).\n",
+     "Confirm that is correct.", test_dual_progress_bar3},
+    {8, "/progress_window/test_dual_progress_bar_4",
+     "The window must disappear and no new window should appear (check below "
+     "this window just in case).\n",
+     "Confirm that is correct.", test_dual_progress_bar4},
+    {-1, NULL, NULL, NULL, NULL}};
+
+/**
+ * GApplication callbacks
+ */
+
+static void do_startup(GObject *object, gpointer data) {
+  progress_window = sdi_progress_window_new(G_APPLICATION(object));
+
+  // this gsettings key stores the path to the .desktop files of the programs
+  // in the dock. Since we need to test a progress bar in the dock, we need
+  // that list.
+  g_autoptr(GSettings) settings = g_settings_new("org.gnome.shell");
+  GList *applist = g_app_info_get_all();
```

Comment:
Good point. Anyway, I already run everything under valgrind. In fact, this isn't a leak because there was a g_list_free_full() after. But g_autolist() is certainly much more elegant.